File: programming files/rgb_to_hex.java
public class rgb_to_hex {
    public static void main(String[] args) {
        int r = 255;
        int g = 127;
        int b = 0;
        String hexColor = rgbToHex(r, g, b);
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }
    //prints out the RGB values of r, g and b.
    public static String rgbToHex(int r, int g, int b) {
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b);
        //returns the RGB values of r, g, b in hexidecimal format.
    }
}

//Test with RGB color (255, 127, 0) = FF7F00

File: programming files/rgb_to_hex.cc
#include <iostream>
#include <sstream>
#include <iomanip>

// Bugs introduced: Tsering- line 8 and 9, r and g should be swapped.
// Line 14 - should be std::tolower, not std::lowercase LA
std::string rgb_to_hex(int r, int g, int b)
{
    r = std::max(0, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    std::stringstream ss;
    ss << std::lowercase << std::hex << std::setfill('3')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    return ss.str();
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
